HyperVExtension/src/HyperVExtension/Models/HyperVVirtualMachine.cs

```diff
@@ -798,11 +798,21 @@ public override string ToString()
 
     private string OperationErrorString(ComputeSystemOperations operation)
     {
+        if (operation == ComputeSystemOperations.Delete)

```

Comment:
Is the `Delete` case needed because `this` can't be used if the VM is deleted?

Reply:
Its because the value inside the "State" property no longer exists once its deleted. Powershell throws an error if you try to get the state of the vm object once its been deleted. And since we print out the "State" property in the ToString method, it throws an exception. Only logging the name is enough for the delete case so this should be fine